### PR TITLE
Remove BuildVersion.id()

### PR DESCRIPTION
**Pull request by thecoop**
BuildVersion should be entirely opaque, this removes the id() method so you can't do any comparisons between versions

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/env/BuildVersion.java [hunk 1]
```diff
@@ -58,6 +62,11 @@ public abstract class BuildVersion {
      */
     public abstract boolean isFutureVersion();
 
+    /**
+     * Returns this build version in a form suitable for storing in node metadata
+     */
+    public abstract String toNodeMetadata();
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/env/BuildVersion.java:68]: Why is the string version specific to node metadata? Could it just be toString and fromString?

**Review by rjernst — COMMENTED** [review]
Looks good, just one question

File: server/src/main/java/org/elasticsearch/env/BuildVersion.java (hunk 1 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/env/BuildVersion.java:68]: The default `toString` is the version semver, not the id number

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
nit on the commit message:

> BuildVersion should be entirely transparent

I think you _opaque_?

**Comment by thecoop** [comment]
Erm, yes...

**Comment by thecoop** [comment]
@elasticmachine update branch

**Pull request merged**